l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -148,103 +99,76 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         uint256 _l2TxGasLimit,
         uint256 _l2TxGasPerPubdataByte
     ) external payable returns (bytes32 l2TxHash) {
-        l2TxHash = deposit(_l2Receiver, _l1Token, _amount, _l2TxGasLimit, _l2TxGasPerPubdataByte, address(0));
+        // we need to deposit and set allowance, as the sender set the allowance to this contract
+        uint256 amount = _depositFundsToSharedBridge(msg.sender, _l1Token, _amount);
+        require(amount == _amount, "3T"); // The token has non-standard transfer logic
+
+        uint256 mintValue = _l1Token == sharedBridge.l1WethAddress() ? msg.value + _amount : msg.value;
```

Comment:
True, I didn't think of this problem before, it was possible to deposit Weth to the erc20 bridge. We will have to solve the issue. 

This means there are two weth token contracts now on L2, the L2Weth and StandardToken. I think the best way to do it would be to stop L1->Standard token deposits, and for Standard token ->L1 withdrawals we can do StandardToken ->L2Weth withdrawals instead. If the user wants to they can withdraw from L2Weth to L1. 

This also means that during the upgrade we will have to unwrap all weth locked in the erc20 contract.

Reply:
I completed the above mentioned changes